nixos/modules/services/x11/display-managers/lightdm-greeters/mini.nix

```diff
@@ -0,0 +1,251 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  dmcfg = config.services.xserver.displayManager;
+  ldmcfg = dmcfg.lightdm;
+  cfg = ldmcfg.greeters.mini;
+
+  xgreeters = pkgs.linkFarm "lightdm-mini-greeter-xgreeters" [{
+    path = "${pkgs.lightdm-mini-greeter}/share/xgreeters/lightdm-mini-greeter.desktop";
+    name = "lightdm-mini-greeter.desktop";
+  }];
+
+  miniGreeterConf = pkgs.writeText "lightdm-mini-greeter.conf"
+    ''
+    [greeter]
+    user = ${cfg.user}
+    show-password-label = ${boolToString cfg.showPasswordLabel}
+    password-label-text = ${cfg.passwordLabelText}
+    show-input-cursor = ${boolToString cfg.showInputCursor}
+
+    [greeter-hotkeys]
+    mod-key = ${cfg.hotkeys.modKey}
+    shutdown-key = ${cfg.hotkeys.shutdownKey}
+    restart-key = ${cfg.hotkeys.restartKey}
+    hibernate-key = ${cfg.hotkeys.hibernateKey}
+    suspend-key = ${cfg.hotkeys.suspendKey}
+
+    [greeter-theme]
+    font = ${cfg.theme.font}
+    font-size = ${cfg.theme.fontSize}
+    text-color = "${cfg.theme.textColor}"
+    error-color = "${cfg.theme.errorColor}"
+    background-image = "${ldmcfg.background}"
+    background-color = "${cfg.theme.backgroundColor}"
+    window-color = "${cfg.theme.windowColor}"
+    border-color = "${cfg.theme.borderColor}"
+    border-width = ${toString cfg.theme.borderWidth}px
+    layout-space = ${toString cfg.theme.layoutSpace}
+    password-color = "${cfg.theme.passwordColor}"
+    password-background-color = "${cfg.theme.passwordBackgroundColor}"
+    '';
+
+in
+{
+  options = {
+
+    services.xserver.displayManager.lightdm.greeters.mini = {
+
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to enable lightdm-mini-greeter as the lightdm greeter.
+
+          Note that this greeter starts only the default X session.
+          You can configure the default X session by
+          <option>services.xserver.desktopManager.default</option> and
+          <option>services.xserver.windowManager.default</option>.
+        '';
+      };
+
+      user = mkOption {
```

Comment:
This add a few too many options. Can add an "extraConfig" like the lightdm-gtk-greeter has? Any options in common with lightdm-gtk-greeter are probably worth keeping.

Reply:
I agree. However, mini-greeter requires that in the config file, all the options are explicitly specified; it does not provide default options when they are omitted in the config file. So, unlike GTK greeter, it could be unsafe to introduce `extraConfig`, with which users might forget to specify some options and after `nixos-rebuild switch` they cannot login.

Reply:
I added `extraConfig` and most options can be set there.

Reply:
@matthewbauer Could you review this again?